File: server/lib/recordingName.js
// Canonical recording-filename validator. Use this everywhere a filename arrives
// from a request; do not hand-roll a prefix/suffix check.
//
// The character class excludes '/' and '\', which is the whole point: Express
// decodes %2F in a route param AFTER matching it as one path segment, so a loose
// startsWith('recording-') && endsWith('.webm') check accepts
// 'recording-x%2F..%2F..%2Fseg.webm' and path.join then escapes the recordings
// directory.
const RECORDING_RE = /^recording-[A-Za-z0-9._-]+\.webm$/

export function isRecordingFilename(name) {
  return typeof name === 'string' && RECORDING_RE.test(name)
}

File: server_py/app/lib/recording_name.py
"""Canonical recording-filename validator. Port of server/lib/recordingName.js.

Use this everywhere a filename arrives from a request; do not hand-roll a
prefix/suffix check. The character class excludes '/' and '\\' — the whole
point, since a loose startswith/endswith check would accept a path-traversal
filename like 'recording-x%2F..%2F..%2Fseg.webm'.
"""
import re

RECORDING_RE = re.compile(r"^recording-[A-Za-z0-9._-]+\.webm$")


def is_recording_filename(name) -> bool:
    return isinstance(name, str) and bool(RECORDING_RE.match(name))
